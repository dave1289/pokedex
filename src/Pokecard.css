.Pokecard {
   border: solid black 2px;
   border-radius: 5px;
   width: 150px;
   text-align: center;
   margin: 10px 10px 10px 10px;
   background-color: grey;
   position: relative;
   justify-content: space-between;
}

.Pokecard-name {
   font-size: 1.2rem;
   margin-bottom: 0px;
}