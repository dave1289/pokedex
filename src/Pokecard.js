import React from 'react'
import './Pokecard.css'

const Pokecard = ({id, name, type, bxp}) => {
   var spriteURL = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`
   return (
   <div className='Pokecard' id={id}>
   <h1 className='Pokecard-name'>{name}</h1>
   <br></br>
   <img src={spriteURL} alt={name} className='Pokecard-img'></img>
   <br></br>
   <sub className='Pokecard-type'>Type: {type}</sub>
   <br></br>
   <sub className='Pokecard-xp'>EXP: {bxp}</sub>
   </div>
   )
}

export default Pokecard