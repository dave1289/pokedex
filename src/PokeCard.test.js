import { render, screen } from '@testing-library/react';
import React from "react"
import Pokecard from './Pokecard';


it('renders our pokecards', () => {
   render(<Pokecard />);
});

it('matches snapshot', () => {
   const { asFragment } = render(<Pokecard />)
   expect(asFragment()).toMatchSnapshot()
})