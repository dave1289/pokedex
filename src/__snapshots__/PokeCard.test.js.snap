// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Pokecard"
  >
    <h1
      class="Pokecard-name"
    />
    <br />
    <img
      class="Pokecard-img"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
    />
    <br />
    <sub
      class="Pokecard-type"
    >
      Type: 
    </sub>
    <br />
    <sub
      class="Pokecard-xp"
    >
      EXP: 
    </sub>
  </div>
</DocumentFragment>
`;
